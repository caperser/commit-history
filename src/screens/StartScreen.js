/*StartScreen.js
//Purpose: The start screen will welcome the users and allow
//them to choose a global history of commits or a specified
//Developer: Ericka Capers
//Date: 12/3/2021
*/

//React native Dependencies
import React from 'react';
import { Text, View, ImageBackground, StyleSheet,
Button, TouchableOpacity, Dimensions } from 'react-native';

//grab dimensions of the device being used
var dimensions = {
width: Dimensions.get('window').width,
height: Dimensions.get('window').height
}

//will hold method for pressing button
const onPress = () => {};

export default function StartScreen () {
  return (
    <View style={ styles.container }>
      <ImageBackground source={require('../images/bg.jpg')} resizeMode="cover" style={styles.image}>
      </ImageBackground>
       <View style={styles.content}>
          <View style = {styles.titlesView}>
           <Text style={styles.imageTitleText}>Welcome!</Text>
           <Text style={styles.imageSubTitleText}>Start viewing Git history now.</Text>
            <TouchableOpacity onPress={onPress} style={styles.button}>
              <Text style={styles.buttonText}>View Recent Global Commits</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onPress} style={styles.button}>
              <Text style={styles.buttonText}>Search Specific Repository</Text>
            </TouchableOpacity>
          </View>
       </View>
    </View>
  )
}

//styles for StartScreen.js
const styles = StyleSheet.create({
  titlesView: {
    alignItems: 'center',
    position: 'absolute',
    top: 40,
    left, left, right, bottom: 0,
  },
  content: {
    alignItems: 'center',
    position: 'absolute',
    top, left, left, right, bottom: 0,
    backgroundColor: 'rgba(0,0,0,0.7)'
  },
  imageTitleText: {
    fontSize: dimensions.width < 500 ? 40: 80,
    color: 'white',
    fontWeight: '100',
    fontFamily: 'sans-serif-thin'
  },
  imageSubTitleText: {
    fontSize: dimensions.width < 500 ? 14: 20,
    color: 'white',
    paddingBottom: dimensions.height*.3
  },
  container: {
    flex: 1,
  },
  image: {
    flex: 1,
    justifyContent: "center",
  },
  text: {
    color: "white",
    fontSize: 42,
    lineHeight: 84,
    fontWeight: "bold",
    textAlign: "center",
    backgroundColor: "#000000c0",
  },
  button: {
      width: dimensions.width >= 500?
        dimensions.width*.4 : dimensions.width*.7,
      backgroundColor: "#483766",
      padding: 20,
      borderRadius: 10,
      marginTop: 20,
      alignSelf: "center",
  },
  buttonText: {
      color: "white",
      textAlign: "center"
  }
});