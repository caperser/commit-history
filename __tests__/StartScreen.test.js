//startScreen.test.js
//tests the rendering and button clicking on StartScreen.js
//Developer: Ericka capers
//Last edit date: 12/6/2021

import React from 'react';

import StartScreen from '../src/screens/StartScreen';
import {View, Text} from 'react-native';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const mockedDispatch = jest.fn();

//mocked navigation
jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
  };
});

describe('StartScreen', () => {
  beforeEach(() => {
      mockedDispatch.mockClear();
  });
  it('should render without issues', () => {
    const component = shallow(<StartScreen />);
    expect(toJson(component)).toMatchSnapshot();
  })
});