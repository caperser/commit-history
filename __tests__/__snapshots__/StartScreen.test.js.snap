// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreen should render without issues 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ImageBackground
    resizeMode="cover"
    source={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.7)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 40,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "sans-serif-thin",
            "fontSize": 80,
            "fontWeight": "100",
          }
        }
      >
        Welcome!
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "paddingBottom": 400.2,
          }
        }
      >
        Start viewing commit history now.
      </Text>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#483766",
            "borderRadius": 10,
            "marginTop": 20,
            "padding": 20,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          Search Specific Repository
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;
